### DISCUSSION TITLE
The First Open Source Festival Payment System. Hi ! Whe are TiBillet :)

### DISCUSSION THREAD

**Discussion by Nasjoe** [stustapay/stustapay#398] [General]
Nice to meet you! We are TiBillet and we have been developing a cashless and festival ticketing solution for 8 years under AGPL license :) 
https://github.com/TiBillet/LaBoutik

we're very open to any form of interoperability and sharing of equipment and skills :) Glad to see we're not the only ones!

https://tibillet.org

**Comment by TheJJ** [comment]
Hi Jonas!
Your tool looks great! Is your plan to deploy the system for your whole Island? 😁

We tailored our system needs to [StuStaCulum](https://stustaculum.de), where people have Mifare Ultralight AES wristbands. The area is semi-closed and all booths and points of sale are self-service.

It's hard to say what features you and us could join forces on, but we surely will find something :)

I could imagine that our backend design is quite unique, where we can define an arbitrary tree of products, users, responsibilities, permissions and from there e.g. statistics.
The Android-App is quite unfancy, it displays the products, can take the orders (or topups with an bluetooth ec terminal) and scans the wristband securely.

Everything is very simple-stupid, but elegantly designed to cover all the features one needs at a festival.

What would you say are the main design points of TiBillet? :)

Cheers
  -- JJ

**Comment by Nasjoe** [comment]
Cool! Thanks for your reply!
TiBillet was also built on the needs of a music and surf festival on Reunion Island. Basically, it was just a simple ticketing and cashless service, as you describe.

Today, after having met many different places, such as social and popular third places, the system is looking to move towards local currencies, time currencies and the valorization of volunteer work.

We've also developed an order-taking section, with printers for preparation areas such as restaurant kitchens.

Today, we're very focused on the notion of federation. A card (or a bracelet, like you) is not disposable. It can be kept to visit other places in the area, and can also be used for associative memberships. Like a badge, it can also be used to control access to book fablab and/or co-working spaces.

But above all, it's a free project run by a cooperative that invites all its users to define the roadmap.

I understand you've been focusing a lot on securing RFID reading. You're writing data to the Mifare chips, aren't you? That's one area we haven't explored very much. We just read the first tag id :)

**Comment by TheJJ** [comment]
Our test-app can write to the chip, but in a production setup they are locked in hardware for read only.
We use an authentication key to do a secure exchange, which allows us to verify that its our chip and which id it has securely.

We don't have ticketing features yet :)

Your roadmap sure sounds great - having one open system to manage payments and access has great potential.

How do you acquire users? Is the island community this strong, or why do businesses trust your tools and pay for them?

**Comment by Nasjoe** [comment]
Ok ! I've always wondered how to counteract copying onto RFID magic cards. I've read a lot of interesting docs, I'll have to take the time one day to experiment :)

We built our system on Reunion Island, but today we're a team of 8 to 10 contributors and users all over France, and even a little in Albania :)

We've been lucky enough to be well identified by the cultural community. A lot of them don't want to work with TicketMaster or Weezevent any more, so it helps a bit to network.

Otherwise, our main argument is open source and free and ethical licenses. We're supported by a network of social third-party venues.
Third places are beginning to be structured into regional networks. And they're launching a lot of workshops on the subjects of emancipating associations from GAFAM and other proprietary solutions.

And as we're building TiBillet as a federation, we also have links with local currency networks, which see cashless as a tool for equipping businesses with compatible payment terminals.


The argument of enhancing the value of volunteer work via time currency also deposited on the cashless card, in a network, works quite well with festivals and cultural associations :)


In any case, don't hesitate to test the solution. Everything is available at https://tibillet.org/docs/presentation/demonstration/ and on github :)
And if you'd like to pool your efforts, we'd love to hear from you!
